storage/include/blockchain/direct_kv_block.h

```diff
@@ -0,0 +1,75 @@
+// Concord
+//
+// Copyright (c) 2020 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "bcstatetransfer/SimpleBCStateTransfer.hpp"
+#include "hash_defs.h"
+#include "kv_types.hpp"
+#include "sliver.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <utility>
+
+namespace concord {
+namespace storage {
+namespace blockchain {
+inline namespace v1DirectKeyValue {
+namespace block {
+inline constexpr auto BLOCK_DIGEST_SIZE = bftEngine::SimpleBlockchainStateTransfer::BLOCK_DIGEST_SIZE;
+
+// Creates a block with the user data appended at the end of the returned Sliver. The passed parentDigest buffer must be
+// of size BLOCK_DIGEST_SIZE bytes.
+concordUtils::Sliver create(const concordUtils::SetOfKeyValuePairs &updates,
+                            concordUtils::SetOfKeyValuePairs &outUpdatesInNewBlock,
+                            const void *parentDigest,
```

Comment:
I totally agree. I am thinking something like below is enough for a Digest type as it has a different type as size changes, it supports comparison operators, has size attached and basically has all we need out of the box:

```
template <std::size_t size>
using Digest = std::array<std::uint8_t, size>;
```

However, I prefer changing state transfer so that it uses this type instead internally too. I am thinking we can refactor that in a separate PR.